esp/templates/survey/survey_form.html

```diff
@@ -0,0 +1,125 @@
+    } else if ( window.attachEvent ) {
+        window.attachEvent( 'onload', setupTimeSlots );
+    }
+    
+    //-->
+</script>
+
+<div id="program_form">
+<form method="post" action="{{ request.get_full_path }}">
+{% csrf_token %}
+{% if section %}
+    {% for question in perclass_questions %}
+        {{ question.render|substitute:section }}
+    {% endfor %}
+{% else %}
+
+    {% for question in questions %}{{ question.render|uselist:classes }}{% endfor %}
+
+    {% if perclass_questions %}
+    {% if timeslots %}
+        {% for ts in timeslots %}
+            <table class="fullwidth">
+            <tr>
+                <th>Questions for {% if ts.description %}{{ ts.description }}{% else %}{{ ts }}{% endif %}</th>
+            </tr>
+            </table>
+            {% if not ts.completed %}
+                <table class="fullwidth">
+                <tr>
+                    
+                    <td width="35%">
+                        <label for="id_attendance_{{ ts.id }}"><b>Which class did you attend in this timeblock?</b></label>
+                    </td>
+                    <td>
+                        <select name="attendance_{{ ts.id }}" id="id_attendance_{{ ts.id }}">
+                            <option value="0">[None]</option>
+                            {% for sec in ts.classsections %}
+                                <option value="{{ sec.id }}" {% if sec.selected %}selected="selected"{% endif %}>{{ sec.parent_class.title }}</option>
+                            {% endfor %}
+                        </select>
+                    </td>
+                </tr>
+                </table>
+                <div id="perclass_questions_{{ ts.id }}">
+                {% for question in perclass_questions %}
+                    {{ question.render|substitute:ts }}
+                {% endfor %}
+                </div>
+            {% else %}
+                <table class="fullwidth">
+                <tr><td style="text-align: center;">You have already filled out a survey for a section at this time.</td></tr>
+                </tr>
+                </table>
+            {% endif %}
+        {% endfor %}
+    {% else %}
+        {% for sec in sections %}
+          <table class="fullwidth">
+           <tr>
+                <th>Questions for {{ sec.emailcode }}: {{ sec.title }} ({% for tm in sec.friendly_times %}{% if forloop.counter > 1 %},{% endif %}{{ tm }}{% endfor %})</th>
+            </tr>
+            </table>
+            {% for question in perclass_questions %}
+                {{ question.render|substitute:sec }}
+            {% endfor %}
+        {% endfor %}
+    {% endif %}
+    {% endif %}
+
+{% endif %}
+
+<table class="fullwidth">
+    <tr>
+        <th>Submit</th>
+    </tr>
+    <tr>
+        <td align="center">
+            Click the button below to finish your survey.  Thank you so much for your time!<br /><br />
+            <input type="Submit" class="fancybutton" value="Submit Survey" /><br />
```

Comment:
Could there be separate submit buttons within the full survey page for each section? Then if a student wanted to, they wouldnt have to go to all the separate pages to do one survey at a time, but could fill out a couple classes at a time through the main survey by clicking smaller submit buttons within the survey?

Reply:
Hmm...this would require a major change to how I've made it work here. Also, isn't the whole idea that students would take a survey for each class, then fill out the remainder of the survey afterwards?

Reply:
I think I'm mostly worried about students being confused by the advertising of class vs program surveys, so i wanted to be able to advertise the whole thing while still having the ability to fill out chunks at a time. But I'm definitely open to other ideas, just throwing concerns out for discussion.